lib/autoload/condition.rb

```diff
@@ -0,0 +1,43 @@
+module Condition
+  # Finds all conditionals that depend on the given object.
+  #
+  # @param [Course::Assessment, Course::Achievement] dependent_object An assessment or
+  #   achievement that conditionals depends on
+  # @return [Object] acts_as_conditional Objects that depend on the condition_object
+  def self.conditionals_of(dependent_object)
+    condition_classes_of(dependent_object).map do |condition_name|
+      Course::Condition.find_by_sql(<<-SQL)
+        SELECT * FROM course_conditions cc
+          INNER JOIN course_condition_#{condition_name.demodulize.downcase.pluralize} ccs
+          ON cc.actable_type = '#{condition_name}'
+            AND cc.actable_id = ccs.id
+            AND ccs.#{dependent_object.class.name.demodulize.downcase}_id = #{dependent_object.id}
+        WHERE course_id = #{dependent_object.course_id}
+      SQL
+    end.flatten.map(&:conditional)
+  end
+
+  private
```

Comment:
This doesn't make the module methods private; private only applies to instance methods. If you want, you need to define the methods within the singleton class:

```ruby
class << self
  def conditionals_of
    # ..
  end

  private

  def condition_classes_of...
```